File: mobile_app/lib/models/file.dart
class FileModel {
  final String id;
  final String filename;
  final String categoryId;
  final String? summary;
  final String? highlightedFilename;
  final String userId;
  final DateTime createdAt;

  FileModel({
    required this.id,
    required this.filename,
    required this.categoryId,
    this.summary,
    this.highlightedFilename,
    required this.userId,
    required this.createdAt,
  });

  factory FileModel.fromJson(Map<String, dynamic> json) {
    return FileModel(
      id: json['id'],
      filename: json['original_filename'] ?? '',
      categoryId: json['category_id'] ?? '',
      summary: json['summary'] ?? '',
      highlightedFilename: json['highlighted_filename'] ?? '',
      userId: json['user_id'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}

File: backend/schemas/file.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class KFileResponse(BaseModel):
    id: str
    original_filename: str
    stored_filename: str
    highlighted_filename: Optional[str]
    summary: Optional[str]
    category_id: Optional[str]
    user_id: str
    created_at: datetime

    class Config:
        from_attributes = True 